src/os/bluestore/BlueStore.h

```diff
@@ -945,6 +961,7 @@ class BlueStore : public ObjectStore,
     bluestore_onode_t onode;  ///< metadata stored as value in kv store
     bool exists;              ///< true if object logically exists
 
+    boost::object_pool<Extent> extent_pool{4096,4096};
```

Comment:
> If I understand these parameters correctly - IMO they look too high for many use patterns. This is especially valid for the first parameter - just a single freed onode will use 1K * 8 bytes, wouldn't it?

The memory cost of each `boost::object_pool` in a free, empty `Onode` instance is just a couple of `size_t` + few padding bytes. However, you're perfectly right that we were allocating far too much for non-empty ones. I just pushed the amended version with an additional comment explaining the parameters. FIO shows 39.6 kIOPS after the changes.

> BTW - does it make sense to have it on per-store, not per-onode basis?

Definitely, this is what we need to have if we want employ huge pages. Also I would like to give a try to sequential data structures (that would ultimately take memory from huge-page-based allocator :-) ) for the `Onode`'s stuff. Although pools aggregate multiple instances together, they don't order them which might mislead/disable data prefetchers. We still see huge number of L3 cache misses in `BlueStore::encode_some`.